content/en/tracing/trace_collection/custom_instrumentation/android/otel.md

```diff
@@ -0,0 +1,544 @@
+    .startSpan()
+
+// ... do your logic here ...
+childSpan.end()
+```
+{{% /tab %}}
+{{% tab "Java" %}}
+```java
+final Span childSpan = tracer.spanBuilder("<span_name>")
+    .setParent(Context.current().with(parentSpan)) // make it child of parent span
+    .startSpan();
+
+// ... do your logic here ...
+childSpan.end();
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+7. (Optional) Provide additional attributes alongside your span:
+
+{{< tabs >}}
+{{% tab "Kotlin" %}}
+```kotlin
+tracer.spanBuilder(spanName = "<span_name>").setAttribute(key = "<key_name>", value = <key_value>).startSpan()
+```
+{{% /tab %}}
+{{% tab "Java" %}}
+```java
+tracer.spanBuilder("<span_name>").setAttribute("<key_name>", <key_value>).startSpan();
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+8. (Optional) Attach an error to a span:
+
+{{< tabs >}}
+{{% tab "Kotlin" %}}
+```kotlin
+span.setStatus(StatusCode.ERROR, description = "<error_description>")
+
+// or if you want to set an exception
+
+span.recordException(exception)
+```
+{{% /tab %}}
+{{% tab "Java" %}}
+```java
+span.setStatus(StatusCode.ERROR, "<error_description>")
+
+// or if you want to set an exception
+
+span.recordException(exception)
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+9. (Optional) Add span links to your span:
+
+{{< tabs >}}
+{{% tab "Kotlin" %}}
+```swift
+val linkedSpan = tracer.spanBuilder(spanName = "linked span").startSpan()
+linkedSpan.end()
+
+val spanWithLinks = tracer.spanBuilder(spanName = "span with links")
+    .addLink(spanContext = linkedSpan.spanContext)
+    .startSpan()
+spanWithLinks.end()
+```
+{{% /tab %}}
+{{% tab "Java" %}}
+```Java
+final Span linkedSpan = tracer.spanBuilder("linked span").startSpan();
+linkedSpan.end();
+
+final Span spanWithLinks = tracer.spanBuilder("span with links")
+        .addLink(linkedSpan.getSpanContext())
+        .startSpan();
+spanWithLinks.end();
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+10. (Optional) Add local parent span to the span generated around the OkHttp request in RUM:
+
+You will first need to add the OpenTelemetry OkHttp extension module to your project dependencies:
+
+```groovy
+android {
+    //(...)
+}
+dependencies {
+    implementation "com.datadoghq:dd-sdk-android-okhttp:x.x.x"
+    implementation "com.datadoghq:dd-sdk-android-okhttp-otel:x.x.x"
+    //(...)
+}
+```
+
+Then once we create an `OkHttp` `Request` we could easilly attach a parent span to the request:
+
+{{< tabs >}}
+{{% tab "Kotlin" %}}
+```swift
+val parentSpan = tracer.spanBuilder(spanName = "parent span").startSpan()
+parentSpan.end()
+val request = Request.Builder()
+    .url("<URL>")
+    .addParentSpan(parentSpan)
+    .build()
+```
+{{% /tab %}}
+{{% tab "Java" %}}
+```Java
+final Span parentSpan = tracer.spanBuilder("parent span").startSpan();
+parentSpan.end()
+final Request:request = new Request.Builder()
+    .url("<URL>")
+    .addParentSpan(parentSpan)
+    .build();
+```
+{{% /tab %}}
+{{< /tabs >}}
+
```

Comment:
```suggestion

## Further reading

{{< partial name="whats-next/whats-next.html" >}}
```
We have to add this for the further reading links to display at the bottom of the page.

Reply:
question is, do we need the `Further Reading` here or should we remove it and just left the links below ?

Reply:
I think we should use our standard Further reading format to align with the typical structure of our other pages. There are already links listed in the frontmatter at the top of this page, so those will automatically populate at the bottom using that partial I listed. 😄